main.tf

```diff
@@ -221,7 +221,7 @@ module "postgresql" {
   user_labels                      = var.tags
 
   database_version                 = "POSTGRES_${each.value.server_version}"
-  database_flags                   = each.value.database_flags
+  database_flags                   = values(zipmap(concat(local.base_database_flags.*.name,each.value.database_flags.*.name),concat(local.base_database_flags,each.value.database_flags)))
```

Comment:
Just to explain this better:

If a user defines an external PG server in their their .tfvars without specifying any `database_flags` then the `base_database_flags` will be used. This will set the `max_prepared_transactions` & `max_connections` values for the server to 1024, as recommended by the SAS documentation. See line: https://github.com/sassoftware/viya4-iac-gcp/pull/150/files#diff-9a576c4980b4f226b7ef1042670d564777bbb0ed8ae2d1df33afbd03e2991fcdR91


If the user does specify values for `database_flags` then their values will be merged on top of the `base_database_flags`, example:

User sets `database_flags = [{ name = "max_prepared_transactions", value = "2222" }, { name = "other_custom_flag", value = true }]`
The final flags applied on the PG server will be: `[{ name = "max_connections", value = "1024" }, { name = "max_prepared_transactions", value = "2222" }, { name = "other_custom_flag", value = true }]`


This logic here on line 224 does that, breaking this line down:
```terraform
base_database_flags = [{ name = "max_prepared_transactions", value = "1024" }, { name = "max_connections", value = "1024" }] # from locals.tf
database_flags [{ name = "max_prepared_transactions", value = "2222" }, { name = "other_custom_flag", value = true }] # user values from .tfvars

concat(local.base_database_flags.*.name,each.value.database_flags.*.name)
# Concat list of all name values from both base_database_flags & database_flags
# Call this A
 [
  "max_prepared_transactions",
  "max_connections",
  "max_prepared_transactions",
  "other_custom_flag",
]

concat(local.base_database_flags,each.value.database_flags)
# Concat list of all maps from both base_database_flags & database_flags
# Call this B
[
  {
    "name" = "max_prepared_transactions"
    "value" = "1024"
  },
  {
    "name" = "max_connections"
    "value" = "1024"
  },
  {
    "name" = "max_prepared_transactions"
    "value" = "2222"
  },
  {
    "name" = "other_custom_flag"
    "value" = true
  },
]


(zipmap(A,B)
# constructs a map from a list of keys and a corresponding list of values
# The key reason why this is used Each pair of elements with the same index from the two lists will be used as the key and value of an element in the resulting map. If the same value appears multiple times in keyslist then the value with the highest index is used in the resulting map.
# This removes the max_prepared_transactions from index 0 which was overridden by a user value
# Call this C
{
  "max_connections" = {
    "name" = "max_connections"
    "value" = "1024"
  }
  "max_prepared_transactions" = {
    "name" = "max_prepared_transactions"
    "value" = "2222"
  }
  "other_custom_flag" = {
    "name" = "other_custom_flag"
    "value" = true
  }
}

values(C)
# values takes a map and returns a list containing the values of the elements in that map.
# This transforms it into the format that `database_flags` parameter from postgressql module needs
[
  {
    "name" = "max_connections"
    "value" = "1024"
  },
  {
    "name" = "max_prepared_transactions"
    "value" = "2222"
  },
  {
    "name" = "other_custom_flag"
    "value" = true
  },
]
```

Reply:
Jay, that additional breakdown is useful, it does a good job of explaining how "If the same value appears multiple times in keyslist then the value with the highest index is used in the resulting map."